### PR TITLE
{diskpool} Add new module `az disk-pool`

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
diskpool

File: src/diskpool/azext_diskpool/manual/_client_factory.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+from ..generated._client_factory import cf_diskpool_cl
+
+
+def cf_disk_pool_zones(cli_ctx, *_):
+    return cf_diskpool_cl(cli_ctx).disk_pool_zones
```
Comment (evelyn-ys) [inline src/diskpool/azext_diskpool/manual/_client_factory.py:15]: Why redefine it under `manual` folder?

File: src/diskpool/azext_diskpool/generated/_params.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    with self.argument_context('disk-pool show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('disk_pool_name', options_list=['--name', '-n', '--disk-pool-name'], type=str, help='The name of '
+                   'the Disk Pool.', id_part='name')
+
+    with self.argument_context('disk-pool create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('disk_pool_name', options_list=['--name', '-n', '--disk-pool-name'], type=str, help='The name of '
+                   'the Disk Pool.')
+        c.argument('sku', action=AddSku, nargs='+', help='Determines the SKU of the Disk Pool')
```
Comment (evelyn-ys) [inline src/diskpool/azext_diskpool/generated/_params.py:45]: Why `sku` can be a list type? Can a `disk-pool` have several skus?

**Comment by evelyn-ys** [comment]
Could you remove debug files under `_az_debug` folder?

File: src/diskpool/azext_diskpool/manual/_client_factory.py (hunk 1 above)
Comment (Juliehzl) [reply src/diskpool/azext_diskpool/manual/_client_factory.py]: fixed

**Review by evelyn-ys — APPROVED** [review]
LGTM

**Comment by Juliehzl** [comment]
Remove generated test file as it is not used and Kai will remove generating test capability in next version

**Pull request merged**